projects/plugins/jetpack/modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -64,11 +65,15 @@ public function init_actions() {
 
 		require_once __DIR__ . '/functions.editor-type.php';
 		add_action( 'edit_form_top', 'Jetpack\EditorType\remember_classic_editor' );
-		add_filter( 'block_editor_settings', 'Jetpack\EditorType\remember_block_editor', 10, 2 );
 		add_action( 'login_init', array( $this, 'allow_block_editor_login' ), 1 );
 		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_block_editor_assets' ), 9 );
 		add_action( 'enqueue_block_assets', array( $this, 'enqueue_block_assets' ) );
 		add_filter( 'mce_external_plugins', array( $this, 'add_tinymce_plugins' ) );
+		if ( version_compare( $wp_version, '5.8', '>=' ) ) {
+			add_filter( 'block_editor_settings_all', 'Jetpack\EditorType\remember_block_editor', 10, 2 );
```

Comment:
I don't think that's going to work as well in WordPress 5.8.

In WordPress 5.7, the `block_editor_settings` filter's second parameter was an instance of `WP_Post`. We use it here:
https://github.com/Automattic/jetpack/blob/26db3e436ccca3e16a62d02d95e792a81f38a1e4/projects/plugins/jetpack/modules/wpcom-block-editor/functions.editor-type.php#L32

In WordPress 5.8, the `block_editor_settings_all` filter's second parameter is different: it is now an instance of `WP_Block_Editor_Context`.